solcx/install.py

```diff
@@ -79,6 +112,58 @@ def install_solc(version=None):
     )
     print("solc {} successfully installed at: {}".format(version, binary_path))
 
+def install_solc_pragma(version, install=True):
+    version = version.strip()
+    comparator_set_range = [i.strip() for i in version.split('||')]
+    comparator_regex = re.compile(r'(?P<operator>([<>]?=?|\^))(?P<version>(?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+))')
+    versions_json = requests.get(ALL_RELEASES).json()
+    range_flag = False
+    for version_json in versions_json:
+        for comparator_set in comparator_set_range:
+            comparators = [m.groupdict() for m in comparator_regex.finditer(comparator_set)]
+            comparator_set_flag = True
+            for comparator in comparators:
+                operator = comparator['operator']
+                if not _compare_versions(version_json['tag_name'], comparator['version'], operator):
+                    comparator_set_flag = False
+            if comparator_set_flag:
+                range_flag = True
+        if range_flag:
+            _check_version(version_json['tag_name'])
+            if install:
+                install_solc(version_json['tag_name'])
+            return version_json['tag_name']
+    raise ValueError("Compatible solc version does not exist")
+
+def _compare_versions(v1, v2, operator='='):
+    v1 = v1.lstrip('v')
+    v2 = v2.lstrip('v')
+    v1_split = [int(i) for i in v1.split('.')]
+    v2_split = [int(i) for i in v2.split('.')]
+    if operator == '' or operator == None:
+        operator = '='
+    if operator == '=':
```

Comment:
Using the [operator](https://docs.python.org/3.7/library/operator.html) module you could shorten this block of if statements and make it a lot more readable.

Reply:
I'm not sure how that would shorten this.  It feels like it'd be a one-for-one swap exactly the same length.